framework/src/actions/SetAdaptivityOptionsAction.C

```diff
@@ -33,6 +33,9 @@ SetAdaptivityOptionsAction::validParams()
   params.addParam<MarkerName>(
       "initial_marker",
       "The name of the Marker to use to adapt the mesh during initial refinement.");
+  params.addParam<bool>("project_initial_marker",
+                        true,
+                        "Whether to project initial Marker values onto the refined mesh.");
```

Comment:
can we have more context here?

Is it needed to project to run 2 rounds of adaptivity? so for a single level we dont need to?

Reply:
As far as the framework is concerned, it's never needed.  libMesh doesn't even write a marker to a variable, so it certainly doesn't care; we get our refinement flag from `Elem::refinement_flag`, not a constant `MONOMIAL` coefficient.  MOOSE writes markers to variables, but once the Adaptivity system itself reads the marker there's nothing in a MOOSE run by default that cares afterward.  The efficient thing to do would have been to only write the marker to a variable in the first place in those situations where it's explicitly requested or where another object uses it later.  And I'd bet that even in those situations, "later" is probably just when another Marker uses the output of the first, and we still only need that output before a refinement, not after.

There are two catches, though.  Number 1 is that users can do whatever they want; it's entirely possible for someone to throw on some postprocessor that reads a marker variable, for example, and defaulting to `false` here would break that.  Number 2 is that everything we write to a variable gets written out to restart and visualization output, and defaulting to `false` here basically breaks every `exodiff` test with initial adaptivity.

I'm not actually sure where to put all this info.  It's too much for an `addParam` blurb, but https://github.com/idaholab/moose/blob/next/framework/doc/content/source/base/Adaptivity.md is basically an empty shim at the moment.  Maybe now's the time to fix that, or maybe there's some other doc I'm not finding?

Reply:
https://mooseframework.inl.gov/moose/syntax/Adaptivity/

Reply:
All those exodiffs can be avoided by setting the parameter in the test btw. It s not a concern to me

Reply:
I do think it would be nice for the default to be `true` ... it would be interesting to see how many apps in CI that breaks.

This makes me think of https://github.com/idaholab/moose/pull/12389 where CI essentially beat me into submission, with the result being that a seemingly inferior default stays the default